comp_loinc/build.py

```diff
@@ -1,83 +1,141 @@
-    :param schema_file:
-    :param code_file:
-    :param output:
-    :return:
+    schema_file: Path = typer.Argument(default=DEFAULTS['schema_file.composed'], resolve_path=True, exists=True),
+    composed_classes_data_file: Path = typer.Argument(
+        default=DEFAULTS['composed_classes_data_file'], resolve_path=True, exists=True),
+    output: Path = typer.Argument(default=DEFAULTS['output.composed'], resolve_path=True, writable=True)
+):
+    """Build composed classes.
+    :param schema_file: Path to LinkML `.yaml` file that defines data model for "grouping classes" of LOINC terms, that 
+    is, classes that group sets of LOINC terms into specific categories.
+    :param composed_classes_data_file: Path to...
+    :param output: Path where output will be saved.
+
+    todo: this is just calling a linkml owl cli, should be written as code with python code
     """
     subprocess.call(["linkml-data2owl", "-s", schema_file, composed_classes_data_file, "-o", output])
 
+
 @app.command(name="merge")
 def merge_owl(
-        owl_directory: str,
-        output: str):
-    # TODO: Consider removing the files created from this point on each time this code executes e.g. any file with 'merge_*'
-    files = [f'{owl_directory}{x}' for x in os.listdir(owl_directory) if ".owl" in x]
+    owl_directory: Path = typer.Argument(default=DEFAULTS['owl_directory'], resolve_path=True, exists=True),
+    output: Path = typer.Argument(default=DEFAULTS['output.merge'], resolve_path=True, writable=True)
+):
+    """Merge owl.
+    :param owl_directory: Path to directory where `.owl` files are stored.
+    :param output: Path where output will be saved.
+
+    TODO: Consider removing the files created from this point each time this code executes e.g. any file with 'merge_*'
+    """
+    files = [f'{owl_directory}{x}' for x in os.listdir(owl_directory) if ".owl" in str(x)]
     subprocess.call(["./robot", "merge", "-i"] + " -i ".join(files).split() + ['-o', output])
 
+
 @app.command(name="reason")
 def reason_owl(
-        owl_directory: str,
-        merged_owl: str,
-        owl_reasoner: str,
-        output: str):
-    call_list = ["./robot", "reason", "-r", owl_reasoner, '-i', f"{owl_directory}{merged_owl}", '-o', f"{owl_directory}{output}"]
+    owl_directory: Path = typer.Argument(default=DEFAULTS['owl_directory'], resolve_path=True, exists=True),
+    merged_owl: str = typer.Argument(default=DEFAULTS['merged_owl']),
+    owl_reasoner: str = typer.Argument(default=DEFAULTS['owl_reasoner']),
+    output: Path = typer.Argument(default=DEFAULTS['output.reason'], resolve_path=True, writable=True)
+):
+    """Reason owl.
+    :param owl_directory: Path to directory where `.owl` files are stored.
+    :param merged_owl: Name of the merged OWL file created from the `merge` command.
+    :param owl_reasoner: The name of the OWL reasoner to use.
+    :param output: Path where output will be saved."""
+    inpath = os.path.join(owl_directory, merged_owl)
```

Comment:
@putmantime Bugfix: Using `os.path.join()` will fix a situation where the user doesn't add a `/` to the end of `owl_directory`. It will also work if they do include the `/`.